### PR TITLE
feat(demo): use links instead of buttons

### LINKED ISSUE #149: Extracting the button into its own component
**As a Pillarbox developer**, I would like the button to be extracted as its own component in order to facilitate its reuse in the various pages.

## Acceptance Criteria

- Create a lit component
- https://github.com/SRGSSR/pillarbox-web/issues/150#issue-2063949777

## Hints

To make the button component reusable in as many cases as possible, we could imagine the following structure:

- button
  - title
  - description
  - miscellaneous
    - icon
    - publish date
    - duration

```html
<a>
	<h2>Title<h2>
	<p>description<p>
	<ul>
		<li><i>icon</i><li>
		<li>publish date<li>
		<li>duration<li>
	</ul>
</a>
```

### LINKED ISSUE #150: Use links rather than buttons to facilitate sharing
**As a Pillarbox developer**, I'd like to use a link instead of the button to make sharing easier.

## Acceptance Criteria

- Change button to link

### PR DESCRIPTION
**Pull request by jboix**
## Description

Closes #149
Closes #150

This change refactors all the content buttons of the demo application into links enabling sharing and further navigation options.

## Changes made

- Created a `ContentLinkComponent` that implements a link within the application with the look of the previous buttons but replaces them for links.
- Refactored all buttons on the "Examples," "Search," and "Lists" pages of the demo application to use the newly created component.
- Corrected a regression introduced in the application's router regarding the base url.

### CODE REVIEW

File: demo/src/lists/lists-page.js [hunk 1]
```diff
@@ -133,35 +132,34 @@ export class ListsPage extends LitElement {
...
-                title="${node.title}">
-            <span class="content-btn-title">${node.title}</span>
-            <div class="content-btn-metadata-container">
-                <i class="material-icons-outlined">${node.mediaType === 'VIDEO' ? 'movie' : 'audiotrack'}</i>
-                <span class="content-btn-info">&nbsp;| ${date} | ${duration}</span>
-            </div>
-        </button>
+      <content-link title="${node.title}" href="lists?${this.#toMediaButtonParams(node)}">
+        <i class="material-icons-outlined">${node.mediaType === 'VIDEO' ? 'movie' : 'audiotrack'}</i>
+        <span>&nbsp;| ${date} | ${duration}</span>
```
Comment (amtins) [inline demo/src/lists/lists-page.js:146]: The icon and date/duration isn't rendered without something like
```suggestion
 <div slot="description">
        <i class="material-icons-outlined">${node.mediaType === 'VIDEO' ? 'movie' : 'audiotrack'}</i>
        <span>&nbsp;| ${date} | ${duration}</span>
</div>
```

Comment (amtins) [reply demo/src/lists/lists-page.js:146]: then maybe adding some CSS to align the icon

File: demo/src/search/search-page.js [hunk 2]
```diff
@@ -134,41 +134,33 @@ export class SearchPage extends LitElement {
...
-        <div class="content-btn-metadata-container">
-          <i
-            class="material-icons-outlined">${r.mediaType === 'VIDEO' ? 'movie' : 'audiotrack'}</i>
-          <span class="content-btn-info">&nbsp;| ${date} | ${duration}</span>
-        </div>
-      </button>
+      <content-link title="${r.title}"
+                   href="search?${this.#toQueryParams(r)}">
+          <i class="material-icons-outlined">${r.mediaType === 'VIDEO' ? 'movie' : 'audiotrack'}</i>
+          <span>&nbsp;| ${date} | ${duration}</span>
```
Comment (amtins) [inline demo/src/search/search-page.js:149]: The icon and date/duration isn't rendered without something like
```suggestion
<div slot="description">
          <i class="material-icons-outlined">${r.mediaType === 'VIDEO' ? 'movie' : 'audiotrack'}</i>
          <span>&nbsp;| ${date} | ${duration}</span>
</div>
```

File: demo/src/theme/theme.scss [hunk 3]
```diff
@@ -108,50 +108,16 @@ dialog::backdrop {
...
-  display: flex;
-  align-items: center;
-  width: 100%;
-
-  * {
-    color: var(--color-6);
-    font-weight: var(--font-weight-5);
-    font-size: var(--size-3);
-  }
-}
```
Comment (amtins) [inline demo/src/theme/theme.scss:149]: Maybe we should keep this for the slot `description` ?

File: demo/src/examples/examples-page.js [hunk 4]
```diff
@@ -23,45 +23,22 @@ export class ExamplesPage extends LitElement {
...
-                  ${example.description || example.title}
-                </span>
-                ${when(example.description, () => html`
-                  <div class="content-btn-metadata-container">
-                    <span class="content-btn-info">${example.title}</span>
-                  </div>`)}
-              </button>
+              <content-link title="${example.description || example.title}"
+                            href="examples?${asQueryParams(example)}">
+                ${example.description ? example.title : ''}
```
Comment (amtins) [inline demo/src/examples/examples-page.js:40]: the description isn't rendered without something like
```suggestion
   <span slot="description">
                 ${example.description ? example.title : ''}
   </span>
```

File: demo/src/core/content-link-component.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener('click', this.#onClick);
+  }
+
+  render() {
+    return html`
+      <a href="${this.href}" title="${this.title}" part="a">
+        <span part="title">${this.title}</span>
+        <slot part="description"></slot>
```
[the remark this answers is not in the document]
Comment (jboix) [reply demo/src/core/content-link-component.js]: As discussed the intention was to style the slot as a "container" element. We've finally decided on using `name` as suggested here and introduced the styling through the pseudo element `::slotted` as such :

```css
::slotted([slot="description"]) {
  // style the element injected in the slot
}
```

This allows us to preserve a default style in the web component itself.

**Comment by jboix** [comment]
As discussed the intention was to style the slot as a "container" element. We've finally decided on using `name` as suggested here and introduced the styling through the pseudo element `::slotted` as such :

```css
::slotted([slot="description"]) {
  // style the element injected in the slot
}
```

This allows us to preserve a default style in the web component itself.

**Review by amtins — APPROVED** [review]
Amazing feature thank you! :shipit:

**Pull request merged**